docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -837,15 +831,473 @@ spec:
+
+[source, yaml,subs="attributes,+macros,callouts"]
+----
+spec:
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        output {
+          s3 {
+             id => 'main_s3_output`
+             temporary_directory => "/usr/share/logstash/data/main/main_s3_output<1>
+          }
+        }
+----
+<1> If you are using more than one plugin of the same type, specify a unique location for each plugin to use.
+
+
+
+
+
+
+
+[id="{p}-logstash-working-with-plugins-scaling"]
+=== Scaling {ls} on ECK
+
+{ls} scalability is highly dependent on the plugins in your pipelines. 
+Some plugins can restrict how you can scale out your Logstash deployment, based on the way that the plugins gather or enrich data.
+
+Plugin categories that require special considerations are:
+
+* <<{p}-logstash-agg-filters>>
+* <<{p}-logstash-inputs-data-pushed>>
+* <<{p}-logstash-inputs-local-checkpoints>>
+* <<{p}-logstash-inputs-external-state>>
+
+If the pipeline _does not_ contain any plugins from these categories, you can increase the number of {ls} instances by setting the `count` property in the {ls} resource:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 3
+----
+
+.Horizontal scaling for {ls} plugins
+
+****
+* Not all {ls} deployments can be scaled horizontally by increasing the number of {ls} Pods defined in the {ls} resource. 
+Depending on the types of plugins in a {ls} installation, increasing the number of pods may cause data duplication, data loss, incorrect data, or may waste resources with pods unable to be utilized correctly.
+
+* The ability of a {ls} installation to scale horizontally is bound by its most restrictive plugin(s). Even if all pipelines are using {logstash-ref}/plugins-inputs-elastic_agent.html[`logstash-input-elastic_agent`] or {logstash-ref}/plugins-inputs-beats.html[`logstash-input-beats`] which should enable full horizontal scaling, introducing a more restrictive input or filter plugin forces the restrictions for pod scaling associated with that plugin.
+****
+
+[id="{p}-logstash-agg-filters"]
+==== Filter plugins: aggregating filters
+
+{ls} installations that use aggregating filters should be treated with particular care: 
+
+* They *must* specify `pipeline.workers=1` for any pipelines that use them.
+* The number of pods cannot be scaled above 1.
+
+// Note to authors and reviewers: Is `logstash-filter-*` a real filter???
+
+Examples of aggregating filters include {logstash-ref}/plugins-filters-aggregate.html[`logstash-filter-aggregate`], `logstash-filter-*`, and any {logstash-ref}/plugins-filters-ruby.html[`logstash-filter-ruby`] implementations that perform aggregations.
+
+[id="{p}-logstash-inputs-data-pushed"]
+==== Input plugins: data pushed to the input
```

Comment:
What do you think about as a comparison between the push and pull model:

"events pushed to Logstash" vs "Logstash pulls events"

or "Logstash receives events" vs "Logstash retrieves events"

and maybe a quick explainer of push vs. pull model

Reply:
I like that approach!  That's what I was initially thinking, but wasn't confident enough in my understanding to put it in writing. :-)

Reply:
I tried to work both "pushed to LS" and "LS receives events" into the heading and text, restating for additional context. Please see what you think.